Sources/OktaJWT.swift

```diff
@@ -187,29 +189,45 @@ public struct OktaJWTValidator {
             return key
         }
 
-        var mJWK: [String: String]
-
-        if jwk == nil {
+        guard jwk == nil else {
+            return try self.createRSAKey(jwk!, kid: kid)
+        }
+        
+        do {
             // Set the class's wellKnown object to ensure we validate the JWT based on values pulled from the well-known endpoint
-            guard let wellKnown = RequestsAPI.getDiscoveryDocument(issuer: validatorOptions["iss"] as! String) else {
+            guard let wellKnown = try RequestsAPI.getDiscoveryDocument(issuer: validatorOptions["iss"] as! String) else {
                 throw OktaAPIError.noWellKnown
             }
-
+            
             // Call keys endpoint and find matching kid and create key
             guard let keysEndpoint = RequestsAPI.getKeysEndpoint(json: wellKnown) else {
                 throw OktaAPIError.noJWKSEndpoint
             }
-
-            guard let mKey = Utils.getKeyFromEndpoint(kid: kid, keysEndpoint) else {
+            
+            guard let mKey = try Utils.getKeyFromEndpoint(kid: kid, keysEndpoint) else {
                 throw OktaAPIError.noKey
             }
-
-            mJWK = mKey
-        } else {
-            mJWK = jwk!
+            
+            return try self.createRSAKey(mKey, kid: kid)
+        } catch where error is OktaAPIError {
+          throw error
+        } catch {
+            let nsError = error as NSError
+            
+            switch nsError.code {
+            case
+                NSURLErrorNotConnectedToInternet,
+                NSURLErrorNetworkConnectionLost,
+                NSURLErrorTimedOut:
+                throw OktaAPIError.offline
+            case NSURLErrorCannotConnectToHost:
+                // Throws noWellKnown for backward-compatibility
+                throw OktaAPIError.noWellKnown
```

Comment:
This section looks redundant. I'd suggest either letting this be handled by default, or implement this as a `case` with a `fallthrough`, to make it explicit that we want cannotConnectToHost to throw a different error.

Reply:
`fallthrough` makes sense. I wanted to emphasize on this case because we cannot throw `offline` error as it breaks existing expected behavior (it must throw `.noWellKnow`).